### PR TITLE
refactor(core): use zod to validate schemas

### PR DESCRIPTION
**Pull request by jog1t**
**Before testing, please remove all related cache (especially .opengb/cache.json)**
What's interesting, enabling a schema validation for modules (here: https://github.com/rivet-gg/opengb/blob/338e554d0a3758b7c29ada56f74b384b87c346a4/src/runtime/context.ts#L85 and here https://github.com/rivet-gg/opengb/blob/338e554d0a3758b7c29ada56f74b384b87c346a4/src/runtime/context.ts#L107-L120) causes registry tests to fail, as most of them are reporting fail-positive statuses right now.

### CODE REVIEW

**Comment by NathanFlurry** [comment]
@jog1t you can change `CACHE_VERSION` in `src/build_state/cache.ts`

**Review by NathanFlurry — DISMISSED** [review]
notes:
- remove all uses of tjs (don't worry about deprecated requestConfig)
- export json schema to meta.json (where userConfigSchema was)
- fix lints
- fix `--strict-schemas` flag to be `--no-strict-schemas`
- config/deps.ts is empty
- remove artifacts/build_schema.ts
- revert src/template/script.ts to og behavior
- replace parseAsync with safeParseAsync

File: src/build/schema/serializer.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+	}
+
+	return s.unknown();
+}
+
+// This is a stripped down version of the typescript lib
+// that only contains the types we want to serialize.
+// This is to avoid serializing the entire typescript lib which
+// is huge and contains a lot of types we don't need.
+const tsLib = Deno.readTextFileSync(import.meta.dirname + "/schema_ts_lib.ts");
```
Comment (MasterPtato) [inline src/build/schema/serializer.ts:184]: This likely won't compile in CI, you'll have to add this file to the generated build artifacts and import it from there

**Review by MasterPtato — APPROVED** [review]

**Review by MasterPtato — DISMISSED** [review]
See comments

**Comment by NathanFlurry** [comment]
### Merge activity

* **Jul 17, 12:36 AM EDT**: @NathanFlurry started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/rivet-gg/opengb/448).

**Pull request merged**